### ISSUE TITLE
localtime() cannot be applied to a naive datetime

### ISSUE THREAD

**Issue by stefanistrate** [wagtail/wagtail#6345]
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Cannot edit page after creation when `USE_TZ = False`.

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`.
2. Change `USE_TZ` in settings to `False`.
3. Create a new page and publish it.
4. Go to the admin to edit the page.
5. An error is shown in browser.

```
Environment:


Request Method: GET
Request URL: http://127.0.0.1:8000/admin/pages/3/edit/

Django Version: 3.1
Python Version: 3.8.3
Installed Applications:
['home',
 'search',
 'wagtail.contrib.forms',
 'wagtail.contrib.redirects',
 'wagtail.embeds',
 'wagtail.sites',
 'wagtail.users',
 'wagtail.snippets',
 'wagtail.documents',
 'wagtail.images',
 'wagtail.search',
 'wagtail.admin',
 'wagtail.core',
 'modelcluster',
 'taggit',
 'django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles']
Installed Middleware:
['django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'wagtail.contrib.redirects.middleware.RedirectMiddleware']


Template error:
In template /Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/wagtail/admin/templates/wagtailadmin/shared/last_updated.html, error at line 4
   localtime() cannot be applied to a naive datetime
   1 : {% load i18n wagtailadmin_tags %}
   2 : {% if last_updated %}
   3 : <span title="{{ last_updated }}" data-wagtail-tooltip="{{ last_updated }}" {% if classname %}class="{{ classname }}"{% endif %}>
   4 :     {% if since_text %}{{ since_text }}{% endif %}  {% timesince_last_update last_updated time_prefix=time_prefix %} 
   5 : </span>
   6 : {% endif %}
   7 : 

Traceback (most recent call last):
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/core/handlers/base.py", line 202, in _get_response
    response = response.render()
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/response.py", line 105, in render
    self.content = self.rendered_content
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/response.py", line 83, in rendered_content
    return template.render(context, self._request)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 170, in render
    return self._render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/loader_tags.py", line 150, in render
    return compiled_parent._render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/loader_tags.py", line 62, in render
    result = block.nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/loader_tags.py", line 192, in render
    return template.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 172, in render
    return self._render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/loader_tags.py", line 192, in render
    return template.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 172, in render
    return self._render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 162, in _render
    return self.nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/defaulttags.py", line 312, in render
    return nodelist.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 938, in render
    bit = node.render_annotated(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/base.py", line 905, in render_annotated
    return self.render(context)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/template/library.py", line 192, in render
    output = self.func(*resolved_args, **resolved_kwargs)
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/wagtail/admin/templatetags/wagtailadmin_tags.py", line 587, in timesince_last_update
    time_str = timezone.localtime(last_update).strftime("%H:%M")
  File "/Users/stefan/.virtualenvs/test-wagtail/lib/python3.8/site-packages/django/utils/timezone.py", line 171, in localtime
    raise ValueError("localtime() cannot be applied to a naive datetime")

Exception Type: ValueError at /admin/pages/3/edit/
Exception Value: localtime() cannot be applied to a naive datetime
```

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes / no)

Yes

### Technical details

* Python version: Run `python --version`.
```
Python 3.8.3
```
* Django version: Look in your requirements.txt, or run `pip show django | grep Version`.
```
Version: 3.1
```
* Wagtail version: Look at the bottom of the Settings menu in the Wagtail admin, or run `pip show wagtail | grep Version:`.
```
Version: 2.10
```
* Browser version: You can use https://www.whatsmybrowser.org/ to find this out.
```
Chrome 84
```

**Comment by gasman** [comment]
Marking as 2.10.1. While I don't think we've ever formally supported `USE_TZ = False`, it looks like most of the pre-2.10 test failures (on Postgres, at least) are due to the testing infrastructure being hard-wired to expect timezone-aware datetimes, rather than Wagtail itself not handling naive datetimes - with the one exception being handling of future go-live dates:

```
======================================================================
ERROR: test_publish_with_future_go_live_does_not_set_live_revision (wagtail.core.tests.test_page_model.TestLiveRevision)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/vagrant/.virtualenvs/bakerydemo/lib/python3.6/site-packages/freezegun/api.py", line 735, in wrapper
    result = func(*args, **kwargs)
  File "/vagrant/wagtail/wagtail/core/tests/test_page_model.py", line 687, in test_publish_with_future_go_live_does_not_set_live_revision
    revision.publish()
  File "/vagrant/wagtail/wagtail/core/models.py", line 1658, in publish
    if page.go_live_at and page.go_live_at > timezone.now():
TypeError: can't compare offset-naive and offset-aware datetimes
```

If this has become a general failure in 2.10 (as opposed to an error on one specific feature) then it sounds like this is worth addressing in a 2.10.x patch release.

**Comment by gasman** [comment]
It turns out that the `test_publish_with_future_go_live_does_not_set_live_revision` failure I mentioned above is actually also a bug in the test suite: in normal circumstances the `page.go_live_at > timezone.now()` comparison should be able to rely on both sides being the same type (naive datetime for USE_TZ=False, aware datetime for USE_TZ=True) but the test sets `page.go_live_at` to be an aware datetime, which is incorrect for USE_TZ=False.

Therefore, on all available evidence 2.9.x does work perfectly with USE_TZ, which is all the more reason to consider the new failure to be a regression in 2.10 (and thus appropriate to fix in 2.10.1) :-)
